share/p2p/shrexpush/pubsub.go

```diff
@@ -61,23 +64,10 @@ func (s *PubSub) Start(context.Context) error {
 // * Unregisters all the added Validators
 // * Closes the `ShrEx/Sub` topic
 func (s *PubSub) Stop(context.Context) error {
-	err := s.pubSub.UnregisterTopicValidator(s.pubSubTopic)
-	if err != nil {
-		return err
-	}
-
+	// TODO(@vgonkivs): unregister the topic validator
 	return s.topic.Close()
 }
 
-// AddValidator registers given Validator for EDS notifications (DataHash).
-// Any amount of Validators can be registered.
-func (s *PubSub) AddValidator(validate Validator) error {
-	return s.pubSub.RegisterTopicValidator(s.pubSubTopic,
-		func(ctx context.Context, p peer.ID, msg *pubsub.Message) pubsub.ValidationResult {
-			return validate(ctx, p, msg.Data)
-		})
-}
-
```

Comment:
Why????

Reply:
Just did not want to keep the dead code until subscriber will be implemented

Reply:
returned back

Reply:
```
	err := s.pubSub.UnregisterTopicValidator(s.pubSubTopic)
	if err != nil {
		return err
	}
```

It was removed bc pubsub will return an error if there is no validator registered

Reply:
@walldiss, already started implementing things in DASer relying on AddValidator